spec/rubocop/cop/style/explicit_block_argument_spec.rb

```diff
@@ -266,4 +266,47 @@ def self.my_method(x, y = 42, *args, **options, &block)
       end
     RUBY
   end
+
+  it 'registers an offense and corrects when there are two methods with the same implementation and name' do
+    expect_offense(<<~RUBY)
+      def foo
+        bar { |baz| yield baz }
+        ^^^^^^^^^^^^^^^^^^^^^^^ Consider using explicit block argument in the surrounding method's signature over `yield`.
+      end
+
+      def foo
+        bar { |baz| yield baz }
+        ^^^^^^^^^^^^^^^^^^^^^^^ Consider using explicit block argument in the surrounding method's signature over `yield`.
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      def foo(&block)
+        bar(&block)
+      end
+
+      def foo(&block)
+        bar(&block)
+      end
+    RUBY
+  end
+
+  it 'registers an offense when there are nested method definitions' do
+    expect_offense(<<~RUBY)
+      def foo
+        def bar.baz
+          bat { |xyz| yield xyz }
+          ^^^^^^^^^^^^^^^^^^^^^^^ Consider using explicit block argument in the surrounding method's signature over `yield`.
+        end
+      end
+    RUBY
+
+    expect_correction(<<~RUBY)
+      def foo
+        def bar.baz(&block)
+          bat(&block)
+        end
+      end
+    RUBY
```

Comment:
That just my two cents. The variable name `bat` might feel a bit abrupt. It could be better to replace it with a metasyntactic variable like the following, for example.

```suggestion
    expect_offense(<<~RUBY)
      def foo
        def bar.baz
          qux { |quux| yield quux }
          ^^^^^^^^^^^^^^^^^^^^^^^ Consider using explicit block argument in the surrounding method's signature over `yield`.
        end
      end
    RUBY

    expect_correction(<<~RUBY)
      def foo
        def bar.baz(&block)
          qux(&block)
        end
      end
    RUBY
```